src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,364 @@
+    # - rels_mondo_non_source_temp: Parent and/or child not from source
+    # - rels_mondo_non_synced_temp: Parent and/or child Mondo ID exists in mondo.db but not mondo.sssom.tsv
+    for x in rels_mondo_mondo:
+        try:
+            rels_mondo_source.add((mondo_source_map[x[0]], x[1], mondo_source_map[x[2]]))
+        except KeyError:
+            try:
+                rels_mondo_non_source_temp.add((mondo_all_sources_map[x[0]], x[1], mondo_all_sources_map[x[2]]))
+            except KeyError:
+                rels_mondo_non_synced_temp.add(x)
+    t1 = datetime.now()
+    print(f'  Completed in {(t1 - t0).seconds} seconds')
+
+    # Determine hierarchy diferences -----------------------------------------------------------------------------------
+    # - find which edges appear in both Mondo and source, or only in one or the other
+    in_both_source_ids_set: Set[RELATIONSHIP] = rels_source.intersection(rels_mondo_source)
+    in_source_only_source_ids_set: Set[RELATIONSHIP] = rels_source.difference(rels_mondo_source)
+    in_mondo_only_source_ids_set: Set[RELATIONSHIP] = rels_mondo_source.difference(rels_source)
+
+    # Create outputs ---------------------------------------------------------------------------------------------------
+    robot_subheader = [{
+        'subject_mondo_id': 'ID',
+        'subject_mondo_label': 'LABEL',
+        'object_mondo_id': 'SC %',
+        'subject_source_id': '',  # >A oboInOwl:source?
+        'object_source_id': '',  # ?
+        'object_mondo_label': '',  # ?
+    }]
+    # - Add metadata to edges (labels, both source/Mondo IDs)
+    in_both: List[Dict] = _add_edge_metadata(in_both_source_ids_set, source_data_map)
+    in_mondo_only: List[Dict] = _add_edge_metadata(in_mondo_only_source_ids_set, source_data_map)
+    in_source_only: List[Dict] = _add_edge_metadata(in_source_only_source_ids_set, source_data_map)
+
+    # Case 3: SCR is direct in the source, but not at all in Mondo
+    df3 = pd.DataFrame(in_source_only)
+    if len(df3) == 0:
+        df3 = pd.DataFrame(columns=list(robot_subheader[0].keys()))
+    df3 = df3.sort_values(['object_mondo_id'], ascending=[False])
+    unmigrated_terms = set(df3[df3['object_mondo_id'] == None]['object_source_id'])
+    if len(unmigrated_terms) >= 200:
+        print(f'Warning: {len(unmigrated_terms)} unmigrated source terms found; a higher number than expected.',
+              file=sys.stderr)
+    df3 = pd.concat([pd.DataFrame(robot_subheader), df3])
+    df3.to_csv(outpath_added, sep='\t', index=False)
+
+    # Case 1: SCR direct in source and Mondo
+    df1 = pd.DataFrame(in_both)
+    if len(df1) == 0:
+        df1 = pd.DataFrame(columns=list(robot_subheader[0].keys()))
+    df1 = df1.sort_values(['subject_mondo_id', 'object_mondo_id'], ascending=[True, True])
+    df1 = pd.concat([pd.DataFrame(robot_subheader), df1])
+    df1.to_csv(outpath_confirmed, sep='\t', index=False)
+
+    # Case 5: SCR is direct in Mondo, and not in the source at all
```

Comment:
##  Case 5: SCR is direct in Mondo, and not in the source at all @matentzn 

<details><summary>Details</summary>
<p>
@matentzn Review this and provide feedback.

### Questions

### FYIs
The file you'll want to look at is `sync-subClassOf.in-mondo-only.tsv`. The columns are:
There is 1 row for each subclass relation in Mondo.
- `subject_id` (Mondo)
- `object_id` (Mondo)
- `subject_label`
- `object_label`
- n `in_SOURCE` columns
- `in_no_source`: This is the primary field of interest in this report.

I changed implementation from what we were thinking of at the last meeting, which was: (a) fully running this at 1 time, completely separate from the other sync subClassOf scripting, to (b) I generate a temporary  `SOURCE.subclass.in-mondo-only.tsv` during the normal process. Then I join them together as `sync-subClassOf.in-mondo-only.tsv`, and finally delete those original source tables. 

</p>
</details>

Reply:
Very interesting work!

> there are actually no subclass relations in Mondo that aren't mirrored in at least one source.

This is technically possible, but hard to believe! But, as you say, if it was true and not due to a bug, I would be very happy.

> 1: Do you really want an output for (i) "Case 5: SCR is direct in Mondo, and not in the source at all" (which is what is written in [the google doc](https://docs.google.com/document/d/1H8fJKiKD-L1tfS-2LJu8t0_2YXJ1PQJ_7Zkoj7Vf7xA/edit))? I figured what you meant was (ii) "Case 5: SCR is direct in Mondo, and either indirect or not at all in source". Was I correct? What would you like?:

This is the correct interpretation of Case 5:

Case 5: SCR is direct in Mondo, and Neither indirect Nor direct in the source.

The point here is to, basically, identify case for subclass relationships that are not at all supported by any of the external resources.

Tables sounds great as you made it, with the `in_source` columns as well. Can you share on Google sheets and share in mondo-ingest channel?

Reply:
@matentzn Glad you like the format. You've confirmed that how you originally wrote the case was correct. That's not what I've done so far, but I can soon (possibly by Monday). After that I will share in Google Sheets and Slack!

Reply:
FYI this is not that important. If this takes time don't let this hold up.

Reply:
Wrapped this up. I created a [slack thread](https://monarchinitiative.slack.com/archives/G01JACCEDAB/p1702253993583929). Here's a [google sheet](https://docs.google.com/spreadsheets/d/1eQx1v5_5oE82GULOSeCO_EeN9rC-2tCaKWEWcBXFMG8/edit#gid=504685640) of results.